flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -1180,6 +1181,14 @@ public final CloseableRegistry getCancelables() {
 			}
 		}
 
+		// Clones the partition to avoid multiple stream edges sharing the same stream partitioner,
+		// like the case of https://issues.apache.org/jira/browse/FLINK-14087.
+		try {
+			outputPartitioner = InstantiationUtil.clone(outputPartitioner, Thread.currentThread().getContextClassLoader());
```

Comment:
In my opinion we should clone the partitioner before initializing it. I think the `configure` can theoretically initialize some state that is not serializable.

Reply:
Very thanks! I agree with you that we should clone the partitioner before configuration.